### PR TITLE
#170947598 setup react router

### PR DESCRIPTION
**Pull request by Niyonsengaeric**
#### What does this PR do?
- Setup react router  (React-router-dom)
#### Description of Task to be completed?

- install react-router-dom package.
- configure the router.
- creating a dummy route for login.
- creating a dummy route for home to test the configuration.
- Create a components and views file structure.

#### How should this be manually tested?

- run `git clone https://github.com/andela/knights-bn-frontend.git` to clone the repo.
- run `cd knights-bn-frontend` in your terminal to enter into the repository.
- run `git fetch` to downloads commits, files, and refs from a remote repository into your local repo.
- run `git checkout ch-Setup-react-router-170947598` to switch to this branch.
- run `npm install` to install dependencies.
- run `npm run start` to start the server.
- The home page will one in your browser.
- You can navigate through the menu to access different pages `login`  and `home` 

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947598](https://www.pivotaltracker.com/story/show/170947598)
#### Screenshots 

                       Dummy  Home page

Screen Shot 2020-04-17 at 17 45 47

                     Dummy  Login Page

Screen Shot 2020-04-17 at 12 12 05

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
-  }
+  "presets": [
+    [
+      "@babel/preset-env",
+      {
+        "modules": "commonjs"
+      }
+    ],
+    "@babel/preset-react",
+    "@babel/preset-typescript"
```
Comment (T2Wil) [inline .babelrc]: is typescript necessary?

File: package.json [hunk 2]
```diff
@@ -38,8 +38,13 @@
     "webpack-dev-server": "^3.10.3"
   },
   "dependencies": {
+    "@babel/plugin-proposal-class-properties": "^7.8.3",
     "@babel/polyfill": "^7.8.7",
+    "@babel/preset-typescript": "^7.9.0",
```
Comment (T2Wil) [inline package.json]: we won't use this package.

File: package.json [hunk 3]
```diff
@@ -38,8 +38,13 @@
...
+    "prettier": "^2.0.4",
```
Comment (T2Wil) [inline package.json:49]: is prettier still useful when we have eslint in place?

File: src/index.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
```
Comment (T2Wil) [inline src/index.html]: As we will be using Bootstrap 4, I think this won't be necessary.

File: webpack.config.js [hunk 5]
```diff
@@ -1,36 +1,40 @@
+const path = require('path');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 const ImagesConfigWebpackPlugin = require('image-config-webpack-plugin');
 const JsConfigWebpackPlugin = require('js-config-webpack-plugin');
 const ScssConfigWebpackPlugin = require('scss-config-webpack-plugin');
-const path = require('path');
 
 module.exports = {
-  entry: ['@babel/polyfill', './src/index.js'],
```
Comment (T2Wil) [inline webpack.config.js]: Why have you deleted this?
@babel/polyfill converts all the latest JS syntax found in '.src/index.js' to the supported version of the running environment.

File: webpack.config.js [hunk 6]
```diff
@@ -1,36 +1,40 @@
...
-  resolve: {
```
Comment (T2Wil) [inline webpack.config.js]: this has not to be deleted too, it tells webpack where to find `@babel/polyfill`

File: webpack.config.js [hunk 7]
```diff
@@ -1,36 +1,40 @@
...
-          },
+        test: /\.(css|mjs|js|jsx)$/i,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+
+        options: {
+          presets: [
+            '@babel/preset-env',
+            {
+              plugins: ['@babel/plugin-proposal-class-properties'],
```
Comment (T2Wil) [inline webpack.config.js]: to keep code clean, move this plugin in the array of plugins at the bottom.

File: .babelrc (hunk 1 above)
Comment (Niyonsengaeric) [reply .babelrc]: thx @T2Wil  let me remove it

File: package.json (hunk 2 above)
Comment (Niyonsengaeric) [reply package.json]: let me also remove it in the packages

File: package.json (hunk 3 above)
Comment (Niyonsengaeric) [reply package.json:49]: they perform different task 

- Prettier is an opinionated code formatter
- ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code.

File: webpack.config.js (hunk 5 above)
Comment (Niyonsengaeric) [reply webpack.config.js]: my bad!

File: webpack.config.js (hunk 7 above)
Comment (Niyonsengaeric) [reply webpack.config.js]: cool

File: webpack.config.js (hunk 6 above)
Comment (Niyonsengaeric) [reply webpack.config.js]: cool @T2Wil , I didn't know when I have deleted it

File: src/views/Home.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import '../styles/styles.css';
+
+function CountdownTimer() {
```
Comment (Kagorora) [inline src/views/Home.js]: @Niyonsengaeric  Please mind code climate errors, consider refactoring

**Review by Kagorora — CHANGES_REQUESTED** [review]
@Niyonsengaeric  Please fix the codeClimate error that I have mentioned

File: src/views/Home.js (hunk 8 above)
Comment (Niyonsengaeric) [reply src/views/Home.js]: solved!

**Review by Kagorora — APPROVED** [review]

**Review by EugeneMunya — APPROVED** [review]

**Review by T2Wil — APPROVED** [review]
Nice work @Niyonsengaeric

**Review by kagabof — APPROVED** [review]
Well done.

**Pull request merged**